### PR TITLE
Update CodeDom TempFileCollection to match .net API

### PR DESCRIPTION
**Pull request by marek-safar**
Missing method is documented here https://msdn.microsoft.com/en-us/library/system.codedom.compiler.tempfilecollection.delete(v=vs.110).aspx

### CODE REVIEW

**Comment by stephentoub** [comment]
Thanks, @marek-safar. This will need to be added to the ref as well:
https://github.com/dotnet/corefx/blob/master/src/System.CodeDom/ref/System.CodeDom.cs#L1202
And can you please also add a test?
https://github.com/dotnet/corefx/blob/master/src/System.CodeDom/tests/CodeCollections/TempFileCollectionTests.cs

**Comment by stephentoub** [comment]
@danmosemsft, do you know why this wasn't flagged by any tools?  Is it because CodeDom isn't part of net standard and that's the only part your tools were looking at?

**Comment by weshaggard** [comment]
>@danmosemsft, do you know why this wasn't flagged by any tools? Is it because CodeDom isn't part of net standard and that's the only part your tools were looking at?

Yes it is because it isn't part of the standard. We did have a net461 apicompat baseline file which would have showed this but it was very noisy and we removed it.

**Comment by danmosemsft** [comment]
@weshaggard so what we need here is a diff between desktop and core, but only for types that are in core? Is there an easy way to generate that? 
@terrajobst How important is to to us that Core has every member in every type that we share with desktop?

File: src/System.CodeDom/tests/CodeCollections/TempFileCollectionTests.cs [hunk 1]
```diff
@@ -248,6 +248,34 @@ public void CopyTo()
...
+                    collection.Delete();
+                    Assert.False(File.Exists(filePath1));
+                    Assert.Equal(0, collection.Count);
+                }
+            }
+            finally
+            {
+                if (File.Exists(filePath1))
+                {
+                    File.Delete(filePath1);
```
Comment (danmosemsft) [inline src/System.CodeDom/tests/CodeCollections/TempFileCollectionTests.cs]: Nit, I don't think you need the .Exists, as .Delete should skip missing files.

**Comment by weshaggard** [comment]
>@weshaggard so what we need here is a diff between desktop and core, but only for types that are in core? Is there an easy way to generate that?

There really isn't any clean way to do generate that as the assembly factoring doesn't match. We can do a whole baseline but it will have a lot of the noise. Folks can use that baseline and look at the differences in the area they are working on. 

>@terrajobst How important is to to us that Core has every member in every type that we share with desktop?

I would say i is pretty important the primary reason we are adding them is to be compatible with desktop. We can still make a call about not including some APIs for a good reason but we shouldn't not include them without a reason.

**Comment by weshaggard** [comment]
I forgot that we moved the baseline file not removed it. We moved it to https://github.com/dotnet/corefx/blob/master/src/shims/ApiCompatBaseline.netfx461.txt. @marek-safar some of your changes are actually removing stuff from that baseline file. Please include it with your PR.

**Comment by marek-safar** [comment]
@weshaggard done

**Comment by danmosemsft** [comment]
Thanks @marek-safar !

**Pull request merged**